WeatherPortal/weatherapp/views.py

```diff
@@ -1,13 +1,48 @@
 import requests
+import os
+import dotenv
+
 from django.contrib.auth import login
 from django.shortcuts import render, redirect
 from django.urls import reverse
+from pprint import pprint
 
 from .forms import CityForm, CustomUserCreationForm
-from .models import City
+from .models import City, Location
+
+API_KEY = os.getenv('API_KEY')
+
+
+def get_city_location(city_name):
+    try:
+        response_city_coords = requests.get(
+            f'http://api.openweathermap.org/geo/1.0/direct?q={city_name}&limit=9&appid={API_KEY}'
+        )
+        get_coords_data = response_city_coords.json()
+        if get_coords_data:
+            return get_coords_data
+        else:
+            return None
```

Comment:
co to znaczy `if get_coords_data`, czy jesteśmy już pewnie, że masz koordynaty?

Reply:
nie, ale sprawdzam to w tym miejscu:
`            get_location_for_new_city = get_city_location(new_add_city)
            if get_location_for_new_city is None:  # check that city exist in geo api
                error_msg_add_city_button = 'No matching result for specific location'`